KomaMRIBase/src/sequences/RF/design.jl

```diff
@@ -0,0 +1,152 @@
+"""
+"""
+function block_pulse(flip_angle; duration=0, freq_offset=0, phase_offset=0,
+    time_bw_product=0, bandwidth=0, delay=0,
+    dead_time=0, ring_down_time=0, sys=nothing)
+
+    if !isnothing(sys)
+        dead_time = sys.RF_dead_time_T
+        ring_down_time = sys.RF_ring_down_T
+    end
+
+    if duration == 0
+        if time_bw_product > 0
+            duration = time_bw_product / bandwidth
+        elseif bandwidth > 0
+            duration = 1 / (4 * bandwidth)
+        else
+            @error "Either bandwidth or duration must be defined"
+        end
+    end
+    amplitude = flip_angle / (2π * γ * duration) * cis(phase_offset)
+
+    if dead_time > delay
+        delay = dead_time
+    end
+
+    dly = Delay(0)
+    if ring_down_time > 0
+        dly = Delay(delay + duration + ring_down_time)     # I NEED a review
+    end
+
+    return RF(amplitude, duration, freq_offset, delay), dly
+end
+
+"""
+"""
+function sinc_pulse(flip_angle; duration=0, freq_offset=0, phase_offset=0,
+    time_bw_product=0, apodization=0.5, centerpos=0.5, delay=0, slice_thickness=0,
+    dead_time=0, ring_down_time=0, Δtrf=1e-5, Δtgr=1e-3, sys=nothing)
+
+    if !isnothing(sys)
+        dead_time = sys.RF_dead_time_T
+        ring_down_time = sys.RF_ring_down_T
+        Δtrf = sys.RF_Δt
+        Δtgr = sys.GR_Δt
+    end
+
+    if duration <= 0
+        @error "rf pulse duration must be positive"
+    end
+    if Δtrf <= 0
+        @error "the Δtrf gradient raster time must be positive"
+    end
+
+    BW = time_bw_product / duration
+    N = Integer(ceil(duration / Δtrf))
+    t = range(0, duration; length=N)
+    window = (1 - apodization) .+ apodization * cos.(2π * ((t  .- (centerpos * duration)) / duration))
+    signal = window .* sinc.(BW * (t  .- (centerpos * duration)))
+    flip = 0.5 * sum(signal[2:end] + signal[1:end-1]) * Δtrf * 2π
+    signal = signal * flip_angle / flip * cis(phase_offset)
+    if dead_time > delay
+        delay = dead_time
+    end
+	rf = RF(signal, duration, freq_offset, delay)
+
+    @assert slice_thickness > 0 "slice_thickness must be provided"
+
+    amplitude = BW / slice_thickness
+    area = amplitude * duration
+    gz = trapezoid(; flat_time=duration, flat_area=area, sys=sys);
+    gz_area = gz.A * (gz.T + gz.rise / 2 + gz.fall / 2)
```

Comment:
we need a function to calculate the area of a Grad (get_kspace almost does this)